pkgs/development/tools/global-platform-pro/default.nix

```diff
@@ -0,0 +1,89 @@
+{ stdenv, fetchFromGitHub, jdk, maven, writeText, makeWrapper, jre_headless, pcsclite }:
+
+# TODO: This is quite a bit of duplicated logic with gephy. Factor it out?
+stdenv.mkDerivation rec {
+    pname = "global-platform-pro";
+    version = "0.3.10-rc11";
+    describeVersion = "v0.3.10-rc11-0-g8923747"; # git describe --tags --always --long --dirty
+    name = "${pname}-${version}";
+
+    src = fetchFromGitHub {
+        owner = "martinpaljak";
+        repo = "GlobalPlatformPro";
+        rev = "v${version}";
+        sha256 = "0rk81x2y7vx1caxm6wa59fjrfxmjn7s8yxaxm764p8m2qxk3m4y2";
+    };
+
+    patches = [ (writeText "${name}-version.patch" ''
```

Comment:
What does this do? Is there an upstream report for it? I think it would be better to put the patch into a separate file. Then add a comment describing what it fixes and a link to upstream.

Reply:
That's not something that can be upstreamed, unfortunately. It's basically bypassing the build system, that attempts to run `git describe --tags --always --long --dirty` which fails due to `fetchFromGitHub` not keeping the git repo.

This patch hardcodes the output based on the `describeVersion` variable, which means it can't really be split into a separate file, or else the `describeVersion` would also have to be in this separate file, and an updater may easily forget to update it when bumping the `version`.

I've written a comment to explain it :)

Reply:
Oh, that makes sense. But while this exact change cannot be upstreamed, we should notify upstream that relying on `git describe` is not ideal. Maybe they'll change it (why do they actually need that?) or at least provide a nicer way to override it at build time.

Generating a patch based on a template seems a bit fishy to me, but I can't think of a better solution either.

Reply:
I can't find the gephy package, could you ping the author?

Reply:
Ugh, it's gephi, not gephy, sorry! I'm not sure on what part of the topic you want me to tag them, though, so I'll let you tag Radvendii where you want :)